File: Analysis/MATLAB/convertpickles.m
function convertpickles(varargin)
% convert fodlers and subfolders of pickles to mat files
% lloyd russell 2016

% select main directory
if ~numel(varargin)
    directory = uigetdir();
else
    directory = varargin{1};
end


% build file list
filelist = glob([directory filesep '**' filesep '*.pkl']);
num_files = numel(filelist);


% convert file list
h = waitbar(0);
for f = 1:num_files
    
    % update progress bar
    waitbar(f/num_files, h, ['Current file: ' strrep(strrep(filelist{f},'\','/'), '_','\_')], 'Interpreter','none');
    
    % do conversion
    pickle2mat('filename',filelist{f}, 'overwrite',false);
    
end
close(h)

File: Analysis/Python/convertpickles.py
import pickle
from scipy.io import savemat
import glob
import os
import time
import sys
try:
	# for Python2
	import Tkinter as tk
	import tkFileDialog as filedialog
except ImportError:
	# for Python3
	import tkinter as tk
	from tkinter import filedialog

def pickle2mat(filename, overwrite=False):
	matname = filename.replace('.pkl', '.mat')
	if not os.path.exists(matname) or (os.path.exists(matname) and overwrite):
		savemat(matname, pickle.load(open(filename, 'rb'), encoding='latin1'))

def update_progress(workdone):
	sys.stdout.write("\rProgress: [{0:50s}] {1:.1f}%".format('#' * int(workdone * 50), workdone*100))

if __name__ == '__main__':
	# select main directory
	root = tk.Tk()
	root.withdraw()
	directory = filedialog.askdirectory()

	# build file list
	filelist = glob.glob(os.path.join(directory, '**', '*.pkl'))
	num_files = len(filelist)

	# convert file list
	for f, filename in enumerate(filelist):
		print('[' + str(f+1) + ' of ' + str(num_files) + '] - ' + filename)
		#update_progress(f/num_files)
		pickle2mat(filename)

	print('Done')